spec/requests/core_induction_programme/core_induction_programme_spec.rb

```diff
@@ -3,18 +3,56 @@
 require "rails_helper"
 
 RSpec.describe "Core Induction Programme", type: :request do
-  describe "GET /core-induction-programmes" do
-    it "renders the core_induction_programmes page" do
-      get "/core-induction-programmes"
-      expect(response).to render_template(:index)
+  let(:core_induction_programme) { create(:core_induction_programme) }
+
+  describe "when an admin user is logged in" do
+    before do
+      admin_user = create(:user, :admin)
+      sign_in admin_user
+    end
+
+    describe "GET /core-induction-programmes" do
+      it "renders the core_induction_programmes page" do
+        get "/core-induction-programmes"
+        expect(response).to render_template(:index)
+      end
+    end
+
+    describe "GET /core-induction-programmes/:id" do
+      it "renders the core_induction_programmes page" do
+        get "/core-induction-programmes/#{core_induction_programme.id}"
+        expect(response).to render_template(:show)
+      end
+    end
+  end
+
+  describe "when an early career teacher is logged in" do
+    before do
+      early_career_teacher = create(:user, :early_career_teacher, { core_induction_programme: core_induction_programme })
+      sign_in early_career_teacher
+    end
+
+    describe "GET /core-induction-programmes" do
+      it "raises an error trying to access core_induction_programme index page" do
+        expect { get "/core-induction-programmes" }.to raise_error Pundit::NotAuthorizedError
+      end
+    end
+
+    describe "GET /core-induction-programmes/:id" do
+      it "renders the core_induction_programme show page" do
+        get "/core-induction-programmes/#{core_induction_programme.id}"
+        expect(response).to render_template(:show)
+      end
     end
```

Comment:
Let's add a test for when there are two programmes, and teacher tries viewing the wrong one.